src/ntfsck.c

```diff
@@ -2439,7 +2439,8 @@ static int ntfsck_scan_index_entries_btree(ntfs_volume *vol)
 					/* TODO: set flags */
 					ie_temp = ntfs_ie_dup_novcn(prev);
 					ret = ntfs_index_rm(ictx);
-					ntfs_ie_add(ictx, ie_temp);
+					if (!ret)
+						ntfs_ie_add(ictx, ie_temp);
```

Comment:
should we check ntfs_ie_add failure case?

Reply:
I will check it on another patch.